code/mission/missionmessage.cpp

```diff
@@ -2045,10 +2086,30 @@ bool filters_match(MessageFilter& filter, ship* it) {
 		    && filter_matches(hud_get_ship_class(it), filter.class_name)
 		    && filter_matches(wing_name, filter.wing_name)
 		    && filter_matches(Ship_info[it->ship_info_index].species, filter.species_bitfield)
-		    && filter_matches(Ship_info[it->ship_info_index].class_type, filter.type_bitfield);
+		    && filter_matches(Ship_info[it->ship_info_index].class_type, filter.type_bitfield)
+		    && filter_matches(it->team, filter.team_bitfield);
 	}
 }
 
+bool outer_filters_match(MessageFilter& filter, int range, ship* sender) {
+	for (auto i: list_range(&Ship_obj_list)) {
+		auto obj = &Objects[i->objnum];
+		if (obj->flags[Object::Object_Flags::Should_be_dead]) {
+			continue;
+		}
+		if ((range >= 0) && (vm_vec_dist(&obj->pos, &Objects[sender->objnum].pos) > range)) {
```

Comment:
\>= ?
I mean, I guess an uninitialized range is set to -1, but is there ever a case where range == 0 is sensible?

Reply:
Since the sender itself is ignored (`if (sender->objnum == i->objnum) { continue; }`), the only way a range of 0 could ever be satisfied is with non-colliding objects in exactly the same position. I don't think it matters in practice, but `> 0`  is probably clearer, so I'm happy to make that change.